### PR TITLE
Move all utility code into `gogolica.core.*` namespace

### LINKED ISSUE #16: Move all our utility namespaces into deeper namespace
The namespaces of generated modules look like `gogolica.storage.v1`, and our utility namespaces are like `gogolica.auth`, `gogolica.utils`, `gogolica.model` and so on.

To distinguish between them, we should consider somehow distinguish our "operational" namespaces from the generated one. I like the names that we currently generate, so maybe we could move the operational modules a level deeper. Something like `gogolica.gen.{auth,utils,model,...}`.

However, `gogolica.gen` is boring, and maybe we should give a nice name to the thing that would be generating the actual user-facing gogolica. I propose either [Viy](https://en.wikipedia.org/wiki/Viy_%28story%29) (more evil), or [Bulba](https://en.wikipedia.org/wiki/Taras_Bulba) (more epic). Then we can have `gogolica.viy` or `gogolica.bulba`.

### PR DESCRIPTION
**Pull request by zudov**
Also updates the README and tests.
Closes #16

### CODE REVIEW

File: src/gogolica/core.clj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 (ns gogolica.core
   (:gen-class)
-  (:require [gogolica.generate :as generate]
-            [gogolica.model    :as model]))
+  (:require [gogolica.core.generate :as generate]
+            [gogolica.core.model    :as model]))
```
Comment (f-f) [inline src/gogolica/core.clj]: Shouldn't they be `gen`?

File: src/gogolica/core/generate.clj [hunk 2]
```diff
@@ -1,14 +1,13 @@
-(ns gogolica.generate
+(ns gogolica.core.generate
   "A namespace for functions that do most of the actual codegen of gogolica."
```
Comment (f-f) [inline src/gogolica/core/generate.clj]: This should also be moved in `gen`

File: src/gogolica/core/generate.clj [hunk 3]
```diff
@@ -1,14 +1,13 @@
...
   (:require [cheshire.core :as json]
             [clojure.string :as str]
             [clojure.set :as set]
             [me.raynes.fs :as fs]
             [clj-http.client :as http]
-            [gogolica.auth :as auth]
-            [gogolica.common :as common]
-            [gogolica.model :as model]
+            [gogolica.core.auth :as auth]
+            [gogolica.core.common :as common]
```
Comment (f-f) [inline src/gogolica/core/generate.clj]: In here we need `model` but we don't need `auth` and `common`

File: src/gogolica/core/model.clj [hunk 4]
```diff
@@ -1,4 +1,4 @@
-(ns gogolica.model
+(ns gogolica.core.model
```
Comment (f-f) [inline src/gogolica/core/model.clj]: `gen`

File: test/gogolica/core_test.clj [hunk 5]
```diff
@@ -1,6 +1,6 @@
 (ns gogolica.core-test
   (:require [clojure.test :refer :all]
-            [gogolica.generate :as g]))
+            [gogolica.core.generate :as g]))
```
Comment (f-f) [inline test/gogolica/core_test.clj]: And here too

**Review by f-f — CHANGES_REQUESTED** [review]

File: src/gogolica/core.clj (hunk 1 above)
Comment (zudov) [reply src/gogolica/core.clj]: Of course :dizzy_face:

File: src/gogolica/core/generate.clj (hunk 3 above)
Comment (zudov) [reply src/gogolica/core/generate.clj]: It wasn't really needed, but it now is (61685ec)

**Comment by zudov** [comment]
Also going to move the tests appropriately

**Review by f-f — APPROVED** [review]

**Comment by f-f** [comment]
Ready to merge?

**Comment by f-f** [comment]
Also afaik this fixes #4 too?

**Comment by zudov** [comment]
> Also afaik this fixes #4 too?

Not yet

**Pull request merged**